scheduler/scheduler-server/src/test/java/functionaltests/job/taskkill/JavaSpawnExecutable.java

```diff
@@ -81,29 +86,49 @@ public void init(Map<String, Serializable> args) throws Exception {
     @Override
     public Serializable execute(TaskResult... results) throws Throwable {
 
-        org.apache.log4j.Logger.getLogger(ProcessTree.class).setLevel(Level.DEBUG);
+        CookieBasedProcessTreeKiller processTreeKiller = null;
         Process process = null;
+        org.apache.log4j.Logger.getLogger(ProcessTree.class).setLevel(Level.DEBUG);
+        try {
+            ProcessBuilder builder = new ProcessBuilder(getNativeExecLauncher(false));
+            builder.directory(getExecutablePath(launchersDir).getParentFile().getCanonicalFile());
+            Map<String, String> environment = builder.environment();
+
+            String cookieSuffix = "JavaExecutable_Job" + getVariables().get(SchedulerVars.PA_JOB_ID) + "Task" +
+                                  getVariables().get(SchedulerVars.PA_TASK_ID);
+
+            // In non-fork mode the child process must be managed by a process tree killer
+            if (!forkMode) {
+                processTreeKiller = CookieBasedProcessTreeKiller.createProcessChildrenKiller(cookieSuffix, environment);
+            }
 
-        process = Runtime.getRuntime().exec(getNativeExecLauncher(false),
-                                            null,
-                                            getExecutablePath(launchersDir).getParentFile().getCanonicalFile());
+            process = builder.start();
 
-        // redirect streams
-        BufferedReader sout = new BufferedReader(new InputStreamReader(process.getInputStream()));
-        BufferedReader serr = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+            // redirect streams
+            BufferedReader sout = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            BufferedReader serr = new BufferedReader(new InputStreamReader(process.getErrorStream()));
 
-        Thread tsout = new Thread(new ThreadReader(sout, System.out));
```

Comment:
u could create a private method like this : 

private void startStream(InputStream processInput, PrintSteam printStream){
    BufferedReader sout = new BufferedReader(new InputStreamReader(processInput));
    Thread ts = new Thread(new ThreadReader(sout, printStream));
    ts.setDaemon(true);
    ts.start();
}

and call it 2 times :  

startStream(process.getInputStream(),System.out );
startStream(process.getErrorStream(),System.err );